test/src/e2e_vm_tests/test_programs/should_pass/stdlib/vec/src/main.sw

```diff
@@ -1810,3 +1811,27 @@ fn test_vector_with_capacity_u64() {
         Option::Some(val) => revert(0), Option::None => (), 
     }
 }
+
+fn test_vector_swap() {
+    let mut vector = ~Vec::new();
+
+    let number0 = 0u8;
+    let number1 = 1u8;
+    let number2 = 2u8;
+    let number3 = 3u8;
+
+    vector.push(number0);
+    vector.push(number1);
+    vector.push(number2);
+    vector.push(number3);
+
+    vector.swap(1, 3);
+
+    match vector.get(1) {
+        Option::Some(val) => assert(val == number3), Option::None => revert(0),
+    }
+
+    match vector.get(3) {
+        Option::Some(val) => assert(val == number1), Option::None => revert(0),
+    }
```

Comment:
There should also be two tests for when `a` and `b` are out of bounds i.e. greater than or equal to the length. you have added in the assertions as preconditions to your method so those conditions must be checked as well